### ISSUE TITLE
docker_container.py: UnboundLocalError: local variable 'post_state' referenced before assignment

### ISSUE THREAD

**Issue by MyIgel** [saltstack/salt#48283]
### Description of Issue
After running salt.apply test=True I get the following Error:
```
----------         
          ID: docker-container-db
    Function: docker_container.running
        Name: db
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1905, in call
                  **cdata['kwargs'])  
                File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1830, in wrapper
                  return f(*args, **kwargs)                                                             
                File "/usr/lib/python2.7/dist-packages/salt/states/docker_container.py", line 2035, in running
                  if pre_state != post_state:
              UnboundLocalError: local variable 'post_state' referenced before assignment
     Started: 15:33:47.895027
    Duration: 4493.225 ms                                            
     Changes:
```

### Setup
Trying to test configure a stopped mariadb container

### Steps to Reproduce Issue
salt.apply

### Versions Report
Master:
```
root@salt:~# salt --versions-report
Salt Version:
           Salt: 2018.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Nov 24 2017, 17:33:09)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1

System Versions:
           dist: debian 9.4
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-6-amd64
         system: Linux
        version: debian 9.4
```

Minion:
```
root@minion:~$ salt-minion --versions-report
Salt Version:
           Salt: 2018.3.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: 1.9.0
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Nov 24 2017, 17:33:09)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1
 
System Versions:
           dist: debian 9.4 
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-6-amd64
         system: Linux
        version: debian 9.4
```

**Comment by jyurdal** [comment]
I have seen the same issue and I have made some investigations.
The reason is that in module docker_container always has __opts__['test'] set to true.
This is not the case for module docker_image.
Does anybody have an idea on why __opts__ are different between modules?

**Comment by jyurdal** [comment]
Actually @garethgreenaway  the issue was fixed in the commit 9515dd17fffc28f7ca9437c51c38a7c41e3fe4bc in salt 2018.3 and higher.
